sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamQueryPlanner.java

```diff
@@ -131,12 +139,42 @@ BeamRelNode convertToBeamRel(String sqlStatement)
               .replace(BeamLogicalConvention.INSTANCE)
               .replace(root.collation)
               .simplify();
-      beamRelNode = (BeamRelNode) planner.transform(0, desiredTraits, root.rel);
-      LOG.info("BeamSQL>\n" + RelOptUtil.toString(beamRelNode));
+
+      // original logical plan
+      RelNode originalRelNode = planner.transform(0, desiredTraits, root.rel);
+
+      // optimized logical plan
+      beamRelNode = optimizeLogicPlan(originalRelNode, desiredTraits);
```

Comment:
I think even though they are separate rulesets, you can just concatenate them all and apply them together in the call to transform above.

Reply:
for here, we can concatenate `CALCITE_LOGICAL_OPTIMIZATIONS` and `BEAM_CONVERSIONS` together and put merged set here

Reply:
I see. will update this.